src/main/java/spoon/reflect/visitor/printer/change/SourceFragment.java

```diff
@@ -0,0 +1,536 @@
+								 */
+								addChild(otherFragment);
+								return otherFragment;
+							}
+						}
+						//the source position of child element is not included in source position of parent element
+						//I (Pavel) am not sure how to handle it, so let's wait until it happens...
+//						if (otherElement instanceof CtAnnotation<?>) {
+//							/*
+//							 * it can happen for annotations of type TYPE_USE and FIELD
+//							 * In such case the annotation belongs to 2 elements
+//							 * And one of them cannot have matching source position - OK
+//							 */
+//							return null;
+//						}
+						//something is wrong ...
+						throw new SpoonException("The SourcePosition of elements are not consistent\nparentFragment: " + parentFragment + "\notherFragment: " + otherFragment);
+					}
+				} else {
+					throw new SpoonException("SourcePosition from unexpected compilation unit: " + otherSourcePosition + " expected is: " + parentFragment.sourcePosition);
+				}
+			}
+			//else these two elements has same instance of SourcePosition.
+			//It is probably OK. Do not created new SourceFragment for that
+			return null;
+		}
+		//do not connect that undefined source position
+		return null;
+	}
+
+	/**
+	 * adds `other` {@link SourceFragment} into tree of {@link SourceFragment}s represented by this root element
+	 *
+	 * @param other to be added {@link SourceFragment}
+	 * @return new root of the tree of the {@link SourceFragment}s. It can be be this or `other`
+	 */
+	public SourceFragment add(SourceFragment other) {
+		if (this == other) {
+			throw new SpoonException("SourceFragment#add must not be called twice for the same SourceFragment");
+			//optionally we might accept that and simply return this
+		}
+		CMP cmp = this.compare(other);
+		switch (cmp) {
+		case OTHER_IS_AFTER:
+			//other is after this
+			addNextSibling(other);
+			return this;
+		case OTHER_IS_BEFORE:
+			//other is before this
+			other.addNextSibling(this);
+			return other;
+		case OTHER_IS_CHILD:
+			//other is child of this
+			addChild(other);
+			return this;
+		case OTHER_IS_PARENT:
+			//other is parent of this
+			other.addChild(this);
+			//merge siblings of `this` as children and siblings of `other`
+			other.mergeSiblingsOfChild(this);
+			return other;
+		}
+		throw new SpoonException("Unexpected compare result: " + cmp);
+	}
+
+	private void mergeSiblingsOfChild(SourceFragment other) {
+		SourceFragment prevOther = other;
+		other = prevOther.getNextSibling();
+		while (other != null) {
+			CMP cmp = compare(other);
+			if (cmp == CMP.OTHER_IS_CHILD) {
+				//ok, it is child too. Keep it as sibling of children
+				prevOther = other;
+				other = other.getNextSibling();
+				continue;
+			} else if (cmp == CMP.OTHER_IS_AFTER) {
+				//the next sibling of child is after `this`
+				//disconnect it from prevOther and connect it as sibling of this
+				prevOther.nextSibling = null;
+				addNextSibling(other);
```

Comment:
When I was analysing the [https://github.com/INRIA/spoon/blob/master/src/main/java/spoon/MavenLauncher.java](MavenLauncher.java) file using this branch, line 331 of this file was throwing an exception. I think there should be a break statement after line 328, otherwise an exception will always be thrown when the control goes into the second if block.

Reply:
you are right. Thank You 👍  I will fix it.

Reply:
fixed

Reply:
it should be fixed with the last commit